src/app/wallet/v2/views/buy_sell_crypto/service_item.nim

```diff
@@ -0,0 +1,65 @@
+import json, strformat, chronicles
```

Comment:
it probably makes more sense to have this object into src/status/wallet ?

Reply:
@alaibe why do you think so? No, this is a view item, and it should be here definitely (it is used in the model and exposed to the qml as is). Items which may be used in the BE may or may not be same as those used in view part. For this particular case since I don't expect that this will be ever used by some other part I forwarded json array form the BE to FE, in every other case I would parsed it in the BE and send it as list of BE items to FE and then map those BE items to FE items used in the model.

Reply:
How it should be... types/structs used in `src/status` should not be used in `src/app` they may be sent from `src/status` and handled by methods in `src/app` which will adapt/convert/map them to types/structs/items for view/models/qml. 

In this branch use case and data flow is like this:
User clicks "Buy/Sell" button -> `src/app` requests list of available services from `src/status` -> `src/status` does async request list of services form `status-go` -> `status-go` sends data to 'src/status' -> `src/status` emits signal to notify `src/app` -> `src/app` converts received data to `CryptoServiceItem` which is added to `CryptoServiceModel` and that model is displayed in qml. That's how it works and that's how it should be.

Not sure what part is wrong or not clear enough?

Reply:
My rationale is the following: If I had to create client  library for status-go I would create an object exactly like this one in my library. Why: if status-go add a new field this is the responsibility of the client library first to add this new field. Then the app using this lib will update accordingly

> types/structs used in src/status should not be used in src/app

I quite strongly disagree with this statement. Rationale: src/status is a dependency of src/app and as such it is normal to leverage object of src/status if there is no need to change them

Here is a few example of the exact rationale I am describing here in our app:
* https://github.com/status-im/status-desktop/blob/master/src/status/profile/profile.nim#L4-L13
* https://github.com/status-im/status-desktop/blob/master/src/status/wallet/account.nim#L27-L34
* https://github.com/status-im/status-desktop/blob/master/src/status/chat/chat.nim#L67-L91
And many more...

Reply:
> If I had to create client library for status-go I would create an object exactly like this one in my library. Why: if status-go add a new field this is the responsibility of the client library first to add this new field. Then the app using this lib will update accordingly

@alaibe you can, and should be like that (I already said that in my first comment [here](https://github.com/status-im/status-desktop/pull/3359#discussion_r701150328)), but that object is important for `src/status` part and when it comes to `src/app` part it should be mapped/converted to appropriate model/items which will be displayed to the user, cause that backend type may contain info which are not relevant for the UI, but important for app's logic.

> I quite strongly disagree with this statement. Rationale: src/status is a dependency of src/app and as such it is normal to leverage object of src/status if there is no need to change them

@alaibe you may agree/disagree, but good coding principles and architectures suggests that, especially in qt/qml where is suggested that each view should have its own corresponding model (containing items), where items may/may not be same as those used in the BE of the app.

> Here is a few example of the exact rationale I am describing here in our app

@alaibe who said this is a good principle? From your examples if we discuss `Chat` type, that is error proned on multiple levels, first all properties are made public so everyone from everywhere may change what he wants, and that is hard for maintaining especially in a bigger projects, second there are many models using `Chat` type where not all properties are relevant for that (that's not only case with `Chat` type, but many more) so why doing like that? So if that `Chat` type handles data coming from `status-go` in the `src/status` the first this which we need is some kind of datastores in that part, and the second if we send some or all of those fields to the `src/app` part, we need to map/convert them to appropriate form which we need to certain view.

Reply:
> @alaibe you may agree/disagree, but good coding principles and architectures suggests that, especially in qt/qml where is suggested that each view should have its own corresponding model (containing items), where items may/may not be same as those used in the BE of the app.

I totally agree with this statement as it diverge from: `types/structs used in src/status should not be used in src/app`
 and I will add: if the model is the same as the backend of the app and doesn't need any transformation, it should be owned by the backend, hence why I think this object should be own (and located in src/app). 


> @alaibe who said this is a good principle? From your examples if we discuss Chat type, that is error proned on multiple levels, first all properties are made public so everyone from everywhere may change what he wants, and that is hard for maintaining especially in a bigger projects, second there are many models using Chat type where not all properties are relevant for that (that's not only case with Chat type, but many more) so why doing like that? So if that Chat type handles data coming from status-go in the src/status the first this which we need is some kind of datastores in that part, and the second if we send some or all of those fields to the src/app part, we need to map/convert them to appropriate form which we need to certain view.

I am not discussing the object implementation itself but his location.

feel free to reach out to me in chat if you prefer to continue this discussion there